crates/rustc_codegen_spirv/src/abi.rs

```diff
@@ -810,5 +810,25 @@ fn trans_intrinsic_type<'tcx>(
                 Err(ErrorReported)
             }
         }
+        IntrinsicType::RuntimeArray => {
+            if ty.size != Size::from_bytes(4) {
+                cx.tcx
+                    .sess
+                    .err("#[spirv(runtime_array)] type must have size 4");
+                return Err(ErrorReported);
+            }
+
+            // We use a generic to indicate the underlying element type.
+            // The spirv type of it will be generated by querying the type of the first generic.
+            if let Some(elem_ty) = substs.types().next() {
+                let element = trans_type_impl(cx, span, cx.layout_of(elem_ty), false);
+                Ok(SpirvType::RuntimeArray { element }.def(span, cx))
```

Comment:
`.def()` produces an array stride decoration, which isn't necessary for opaque resource types ([shader-playground GLSL example](http://shader-playground.timjones.io/a2a0f437a19f698843917b7f3ac21c03)) and is explicitly forbidden for buffers.